<a href="https://colab.research.google.com/github/katemanee006/DADS5001/blob/main/Assignment_Plotly_6610422006.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [21]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import numpy as np


Data source: https://www.kaggle.com/datasets/xontoloyo/data-penjualan-zara?select=zara.csv

In [22]:
zara = pd.read_csv('https://raw.githubusercontent.com/katemanee006/DADS5001/main/zara.csv')
zara.head()

Product ID Product Position Promotion Product Category Seasonal  \
0      185102            Aisle        No         Clothing       No   
1      188771            Aisle        No         Clothing       No   
2      180176          End-cap       Yes         Clothing      Yes   
3      112917            Aisle       Yes         Clothing      Yes   
4      192936          End-cap        No         Clothing      Yes   

   Sales Volume brand                                                url  \
0          2823  Zara  https://www.zara.com/us/en/basic-puffer-jacket...   
1           654  Zara  https://www.zara.com/us/en/tuxedo-jacket-p0889...   
2          2220  Zara  https://www.zara.com/us/en/slim-fit-suit-jacke...   
3          1568  Zara  https://www.zara.com/us/en/stretch-suit-jacket...   
4          2942  Zara  https://www.zara.com/us/en/double-faced-jacket...   

                sku                  name  \
0   272145190-250-2   BASIC PUFFER JACKET   
1  324052738-800-46         TUXEDO JACKET   
2  335342680-800-44  SLIM FIT SUIT JACKET   
3  328303236-420-44   STRETCH SUIT JACKET   
4   312368260-800-2   DOUBLE FACED JACKET   

                                         description   price currency  \
0  Puffer jacket made of tear-resistant ripstop f...   19.99      USD   
1  Straight fit blazer. Pointed lapel collar and ...  169.00      USD   
2  Slim fit jacket. Notched lapel collar. Long sl...  129.00      USD   
3  Slim fit jacket made of viscose blend fabric. ...  129.00      USD   
4  Jacket made of faux leather faux shearling wit...  139.00      USD   

                   scraped_at    terms section  
0  2024-02-19T08:50:05.654618  jackets     MAN  
1  2024-02-19T08:50:06.590930  jackets     MAN  
2  2024-02-19T08:50:07.301419  jackets     MAN  
3  2024-02-19T08:50:07.882922  jackets     MAN  
4  2024-02-19T08:50:08.453847  jackets     MAN

## EDA

In [23]:
zara.isnull().sum()

## ไม่ได้ใช้ข้อมูล name และ description จึงไม่ได้แก้ไขค่า null


Product ID          0
Product Position    0
Promotion           0
Product Category    0
Seasonal            0
Sales Volume        0
brand               0
url                 0
sku                 0
name                1
description         2
price               0
currency            0
scraped_at          0
terms               0
section             0
dtype: int64

In [24]:
zara.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 252 entries, 0 to 251
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Product ID        252 non-null    int64  
 1   Product Position  252 non-null    object 
 2   Promotion         252 non-null    object 
 3   Product Category  252 non-null    object 
 4   Seasonal          252 non-null    object 
 5   Sales Volume      252 non-null    int64  
 6   brand             252 non-null    object 
 7   url               252 non-null    object 
 8   sku               252 non-null    object 
 9   name              251 non-null    object 
 10  description       250 non-null    object 
 11  price             252 non-null    float64
 12  currency          252 non-null    object 
 13  scraped_at        252 non-null    object 
 14  terms             252 non-null    object 
 15  section           252 non-null    object 
dtypes: float64(1), int64(2), object(13)
memory u

In [25]:
zara.columns.values

array(['Product ID', 'Product Position', 'Promotion', 'Product Category',
       'Seasonal', 'Sales Volume', 'brand', 'url', 'sku', 'name',
       'description', 'price', 'currency', 'scraped_at', 'terms',
       'section'], dtype=object)

In [26]:
zara.shape

(252, 16)

In [27]:
## add column total price
zara['totalprice'] = zara['price'] * zara['Sales Volume']

In [28]:
zara.head()

Product ID Product Position Promotion Product Category Seasonal  \
0      185102            Aisle        No         Clothing       No   
1      188771            Aisle        No         Clothing       No   
2      180176          End-cap       Yes         Clothing      Yes   
3      112917            Aisle       Yes         Clothing      Yes   
4      192936          End-cap        No         Clothing      Yes   

   Sales Volume brand                                                url  \
0          2823  Zara  https://www.zara.com/us/en/basic-puffer-jacket...   
1           654  Zara  https://www.zara.com/us/en/tuxedo-jacket-p0889...   
2          2220  Zara  https://www.zara.com/us/en/slim-fit-suit-jacke...   
3          1568  Zara  https://www.zara.com/us/en/stretch-suit-jacket...   
4          2942  Zara  https://www.zara.com/us/en/double-faced-jacket...   

                sku                  name  \
0   272145190-250-2   BASIC PUFFER JACKET   
1  324052738-800-46         TUXEDO JACKET   
2  335342680-800-44  SLIM FIT SUIT JACKET   
3  328303236-420-44   STRETCH SUIT JACKET   
4   312368260-800-2   DOUBLE FACED JACKET   

                                         description   price currency  \
0  Puffer jacket made of tear-resistant ripstop f...   19.99      USD   
1  Straight fit blazer. Pointed lapel collar and ...  169.00      USD   
2  Slim fit jacket. Notched lapel collar. Long sl...  129.00      USD   
3  Slim fit jacket made of viscose blend fabric. ...  129.00      USD   
4  Jacket made of faux leather faux shearling wit...  139.00      USD   

                   scraped_at    terms section  totalprice  
0  2024-02-19T08:50:05.654618  jackets     MAN    56431.77  
1  2024-02-19T08:50:06.590930  jackets     MAN   110526.00  
2  2024-02-19T08:50:07.301419  jackets     MAN   286380.00  
3  2024-02-19T08:50:07.882922  jackets     MAN   202272.00  
4  2024-02-19T08:50:08.453847  jackets     MAN   408938.00

## **Plot**

**อธิบายชุดข้อมูลที่เลือกใช้ :** ชุดข้อมูลนี้แสดงยอดขายของสินค้าแต่ละประเภท ที่จะสะท้อนให้เห็นถึงพฤติกรรมของผู้บริโภค ว่าสินค้าประเภทใดได้รับความนิยม

ข้อมูลที่ต้องการศึกษา: เพื่อเปรียบเทียบยอดขายของสินค้าแต่ละประเภทในร้าน Zara
Terms: ประเภทของสินค้า
Total Price: Sales Volume*Price ยอดขายของสินค้า

In [29]:
# หาผลรวมยอดขายสินค้าแยกตาม terms
price_sum_by_terms = zara.groupby('terms')['totalprice'].sum()
price_sum_by_terms

terms
jackets     26581815.87
jeans         864385.25
shoes        3754837.63
sweaters     4090631.48
t-shirts     3696806.25
Name: totalprice, dtype: float64

In [30]:
fig = go.Figure(go.Waterfall(
    name="terms",
    orientation="v",
    x=price_sum_by_terms.index.to_numpy(),
    textposition="outside",
    text=price_sum_by_terms.to_numpy(),
    y=price_sum_by_terms.diff().fillna(price_sum_by_terms),
    connector={"line": {"color": "rgb(63, 63, 63)"}}
))

fig.update_layout(
    title="Total price by terms of ZARA",
    showlegend=True
)

fig.show()

**กราฟสำหรับใช้ในการวิเคราะห์ :**
เลือกใช้กราฟ Waterfall ในการวิเคราะห์ยอดขายของสินค้าแต่ละ Terms โดยแกน X คือ Terms และ แกน Y คือ ยอดขายสินค้า แบ่งสีตามการเพิ่มขึ้น/ลดลงของยอดขาย ที่เทียบกับกลุ่มก่อนหน้า

**ผลลัพธ์ที่ได้จากกราฟเบื้องต้น :**
*   สินค้าที่ขายได้มากที่สุดคือ เสื้อแจ็กเก็ต(jackets) คิดเป็น 26,581,815.87 USD
*   สินค้าที่ขายได้น้อยที่สุดคือ เสื้อกันหนาว(sweaters) คิดเป็น 3,698,806.25 USD

